sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -190,4 +191,18 @@ object DataSourceUtils {
     case LegacyBehaviorPolicy.LEGACY => RebaseDateTime.rebaseGregorianToJulianMicros
     case LegacyBehaviorPolicy.CORRECTED => identity[Long]
   }
+
+  private[sql] def checkDuplicateOptions(
+      extraOptionsMap: Map[String, String], table: CatalogTable): Unit = {
```

Comment:
we should take care of case sensitivity:
```
def checkConflictingOptions(table: CatalogTable, options: CaseInsensitiveStringMap): Unit = {
  table.storage.properties.foreach { case (k, v) =>
    if (options.get(k) == v) throw ...
  }
}
```

Reply:
Copy that, done in b4aa95e.